x-pack/plugins/security_solution/public/detections/components/rules/step_about_rule/index.test.tsx

```diff
@@ -388,4 +398,24 @@ describe('StepAboutRuleComponent', () => {
     const indexNames = ['.ml-anomalies-shared'];
     expect(useFetchIndex).lastCalledWith(indexNames);
   });
+
+  it('should use default rule index if selected ML jobs are not installed when creating/editing ML rule', async () => {
```

Comment:
does ML rule have index at all?
According to schema, it does not https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/server/lib/detection_engine/rule_schema/model/rule_schemas.ts#L217-L221

Is it possible, default index is not `undefined` for ML rules?
I can see, `useRuleIndices` is called with rule index, but wonder if there any cases it is not `undefined` for ML rule

```ts
  const { ruleIndices } = useRuleIndices(
    defineRuleData?.machineLearningJobId,
    defineRuleData?.index
  );

```